src/azul/service/responseobjects/elastic_request_builder.py

```diff
@@ -499,7 +504,14 @@ def transform_request(self,
             # Apply paging
             es_search = self.apply_paging(es_search, pagination)
             # Execute ElasticSearch request
-            es_response = es_search.execute(ignore_cache=True)
+
+            try:
+                es_response = es_search.execute(ignore_cache=True)
+            except elasticsearch.NotFoundError:
+                raise ElasticsearchIndexNotFoundError(f'Could not find the Elasticsearch index,'
```

Comment:
The message should be hard-coded in the constructor of the exception class. And omit the index name and ES domain from the message. There is no guarantee that `config.es_index_name(entity_type)` is indeed the missing index. You could check whether the instance of NotFoundError contains that information, but if it doesn't, it would be better to not assume what the missing index is.

Reply:
I was able to find the index name in `NotFoundError().info["error"]["index"]` from the exception, but not the ElasticSearch Domain. I'll remove the url from the message.